src/common.rs

```diff
@@ -827,36 +888,36 @@ impl PartialEq for Identifier {
 impl Hash for Identifier {
     fn hash<H: Hasher>(&self, state: &mut H) {
         match self {
-            Identifier::Quoted(a) => a.hash(state),
-            Identifier::Unquoted(a) => a.to_lowercase().hash(state),
+            Identifier::Quoted(a) => a.value.hash(state),
+            Identifier::Unquoted(a) => a.value.to_lowercase().hash(state),
         }
     }
 }
 
 impl Display for Identifier {
     fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
         match &self {
-            Identifier::Quoted(txt) => write!(f, "\"{}\"", txt.replace('\"', "\"\"")),
-            Identifier::Unquoted(txt) => write!(f, "{}", txt),
+            Identifier::Quoted(txt) => write!(f, "\"{}\"", txt.value.replace('\"', "\"\"")),
+            Identifier::Unquoted(txt) => write!(f, "{}", txt.value),
         }
     }
 }
 
 impl Default for Identifier {
     fn default() -> Self {
-        Identifier::Unquoted("".to_string())
+        Identifier::Unquoted("".to_string().into())
     }
 }
 
 impl From<&str> for Identifier {
     fn from(txt: &str) -> Self {
-        Identifier::parse(txt)
+        Identifier::parse(txt, None)
     }
 }
 
 impl From<&String> for Identifier {
```

Comment:
This can be just deleted

Reply:
some test rely on `impl From<&str> for Identifier` too, deleting those tests too????

Reply:
Thats strange, it seems to work fine for me:
<img width="801" alt="image" src="https://github.com/user-attachments/assets/19ac1948-55f7-4647-bd27-c2d0cc6c653d" />

Reply:
what if leave span as Optional more flexible???